pkg/registry/batch/job/strategy.go

```diff
@@ -61,10 +61,9 @@ func (jobStrategy) DefaultGarbageCollectionPolicy(ctx context.Context) rest.Garb
 	}
 	switch groupVersion {
 	case batchv1.SchemeGroupVersion:
-		// for back compatibility
-		return rest.OrphanDependents
-	default:
 		return rest.DeleteDependents
+	default:
+		return rest.OrphanDependents
```

Comment:
For older job api.

Reply:
Revert this bit. There are no older job API objects, and we want the default for any newly introduced versions to be DeleteDependents